app/controllers/user.js

```diff
@@ -44,6 +46,61 @@ exports.create = (req, res, next) => {
   
 };
 
+exports.signin = (req, res, next) => {
+
+  try {
+    validationResult(req).throw();
+  } catch (err) {
+    let errors = {};
+    if(err.mapped().email){
+      errors.email = err.mapped().email.msg + ';';
+    }
+    if(err.mapped().password){
+      errors.password = err.mapped().password.msg;
+    }
+    return res.status(400).send(errors);
+     
+  }
+
+  let input = emptyToNull(req.body);
+
+  if (req.headers.token){
+
+    let token = jwt.decode(req.headers.token, secret);
+
+    if(token.token == input.email){
+      return res.status(200).send('You are already logged in!');
+    }
+    
+  }
+
+  User.findOne({
+    where: {
+      email: input.email
+    }
+  }).then(result => {
+    
+    if(!result){
+      logger.info(`Login attempt with invalid email: "${input.email}"`);
+      return res.status(401).send('Database error');
+    }
+
+    if(!result.validPassword(input.password)){
+      logger.info(`Failed login attempt to account with email "${input.email}", invalid password`);
+      return res.status(401).send('Invalid password');
+    }
+    
+    let token = jwt.encode({token: result.email}, secret);
+    logger.info(`User ${result.email} successfully logged in`);
+    res.status(200).json(token);
```

Comment:
As I told you the other day, here we could send some more information like the user, so the response would look something like these:
```
{
  user: {
    id: ...,
    email: ...,
    ...
  },
  access_token: token
}
```
What do you think?